crates/bevy_render/src/mesh/mesh/mod.rs

```diff
@@ -636,10 +691,21 @@ impl RenderAsset for Mesh {
             index_format: mesh.indices().unwrap().into(),
         });
 
+        let vertex_layout = mesh.vertex_layout();
+
+        let mut hasher = AHasher::default();
+        vertex_layout.hash(&mut hasher);
```

Comment:
I think the RenderPipelineCache should be responsible for handling vertex layout hashing. A method like `insert_vertex_layout` should return a VertexLayoutKey when given a vertex layout. This reduces the complexity for callers and ensures consistent, correct hashes. I'd also prefer it if this lived directly on RenderPipelineCache instead of exposing a public vertex_layout_cache field. Seems simpler / more discoverable.